File: javascript/bob/bob.js
var Bob = function() {
    this.hey = function(message) {
        if (empty(message)) {
            return 'Fine. Be that way!';
        } else if (yelling(message)) {
            return 'Woah, chill out!';
        } else if (question(message)) {
            return 'Sure.';
        }
        return 'Whatever.';
    };

    var question = function(message) {
        return message.charAt(message.length - 1) === '?';
    };

    var yelling = function(message) {
        return message.toUpperCase() === message
            && message.toLowerCase() !== message.toUpperCase();
    };

    var empty = function(message) {
        return message.trim().length === 0
    };
};

File: elixir/bob/bob.exs
defmodule Teenager do
  def hey(input) do
    cond do
      silent?(input)      -> "Fine. Be that way!"
      shouting?(input)    -> "Woah, chill out!"
      questioning?(input) -> "Sure."
      true                -> "Whatever."
    end
  end
  
  defp shouting?(input) do
    String.upcase(input) == input and String.downcase(input) != input
  end
  
  defp silent?(input) do
    String.strip(input) == ""
  end

  defp questioning?(input) do
    String.last(input) == "?"
  end
end